core/src/test/java/com/datastax/oss/driver/internal/core/control/ControlConnectionTest.java

```diff
@@ -155,26 +154,23 @@ public void should_reconnect_if_channel_goes_down() throws Exception {
     CompletionStage<Void> initFuture = controlConnection.init(false, false, false);
     factoryHelper.waitForCall(node1);
 
-    waitForPendingAdminTasks();
-    assertThatStage(initFuture).isSuccess();
-    assertThat(controlConnection.channel()).isEqualTo(channel1);
-    verify(eventBus).fire(ChannelEvent.channelOpened(node1));
+    assertThatStage(initFuture)
+        .isSuccess(v -> assertThat(controlConnection.channel()).isEqualTo(channel1));
+    verify(eventBus, VERIFY_TIMEOUT).fire(ChannelEvent.channelOpened(node1));
 
     // When
     channel1.close();
-    waitForPendingAdminTasks();
 
     // Then
     // a reconnection was started
-    verify(reconnectionSchedule).nextDelay();
+    verify(reconnectionSchedule, VERIFY_TIMEOUT).nextDelay();
     factoryHelper.waitForCall(node1);
     factoryHelper.waitForCall(node2);
-    waitForPendingAdminTasks();
-    assertThat(controlConnection.channel()).isEqualTo(channel2);
-    verify(eventBus).fire(ChannelEvent.channelClosed(node1));
-    verify(eventBus).fire(ChannelEvent.channelOpened(node2));
-    verify(metadataManager).refreshNodes();
-    verify(loadBalancingPolicyWrapper).init();
+    await().untilAsserted(() -> assertThat(controlConnection.channel()).isEqualTo(channel2));
+    verify(eventBus, VERIFY_TIMEOUT).fire(ChannelEvent.channelClosed(node1));
+    verify(eventBus, VERIFY_TIMEOUT).fire(ChannelEvent.channelOpened(node2));
+    verify(metadataManager, VERIFY_TIMEOUT).refreshNodes();
+    verify(loadBalancingPolicyWrapper, VERIFY_TIMEOUT).init();
```

Comment:
I'm being a bit heavy-handed with the verify timeouts, some events could probably piggy-back on previous timed verifications. But I prefer not to rely on subtle reasoning about ordering.